lib/Model/Importer/DebugInfo/PDBImporter.cpp

```diff
@@ -0,0 +1,1089 @@
+    auto PtrSize = getPointerSizeInBytes(Ptr.getPointerKind());
+    std::vector<Qualifier> Qualifiers{ Qualifier::createPointer(PtrSize) };
+    QualifiedType Qty(ProcessedTypes[RefTy], Qualifiers);
+
+    TypeDef->UnderlyingType = Qty;
+
+    auto TyPath = Model->recordNewType(std::move(TypedefTy));
+    ProcessedTypes[CurrentTypeIndex] = TyPath;
+  } else {
+    revng_log(DILogger, "LF_POINTER: Unknown ref type " << RefTy.getIndex());
+  }
+
+  return Error::success();
+}
+
+// Parse LF_ARRAY.
+Error PDBImporterTypeVisitor::visitKnownRecord(CVType &Record,
+                                               ArrayRecord &Array) {
+  using namespace model;
+  auto TypedefTy = makeType<TypedefType>();
+  auto TypeDef = cast<TypedefType>(TypedefTy.get());
+
+  TypeIndex ElementTy = Array.getElementType();
+
+  if (ProcessedTypes.count(ElementTy) == 0 and ElementTy.isSimple())
+    createPrimitiveType(ElementTy);
+
+  if (ProcessedTypes.count(ElementTy)) {
+    auto MaybeSize = ProcessedTypes[ElementTy].get()->size();
+    if (not MaybeSize or *MaybeSize == 0 or Array.getSize() == 0) {
+      return Error::success();
+    }
+
+    const uint64_t ArraySize = Array.getSize() / *MaybeSize;
+    std::vector<Qualifier> Qualifiers{ Qualifier::createArray(ArraySize) };
+    QualifiedType Qty(ProcessedTypes[ElementTy], Qualifiers);
+
+    TypeDef->UnderlyingType = Qty;
+
+    auto TyPath = Model->recordNewType(std::move(TypedefTy));
+    ProcessedTypes[CurrentTypeIndex] = TyPath;
+  } else {
+    revng_log(DILogger,
+              "LF_ARRAY: Unknown element type " << ElementTy.getIndex());
+  }
+
+  return Error::success();
+}
+
+// Parse LF_MODIFIER.
+Error PDBImporterTypeVisitor::visitKnownRecord(CVType &Record,
+                                               ModifierRecord &Modifier) {
+  auto TypedefTy = model::makeType<model::TypedefType>();
+  auto TypeDef = cast<model::TypedefType>(TypedefTy.get());
+
+  TypeIndex RefTy = Modifier.getModifiedType();
+
+  if (ProcessedTypes.count(RefTy) == 0 and RefTy.isSimple())
+    createPrimitiveType(RefTy);
+
+  if (ProcessedTypes.count(RefTy)) {
+    std::vector<model::Qualifier> Qualifiers;
+    auto HasConst = Modifier.getModifiers() & ModifierOptions::Const;
+    if (HasConst != ModifierOptions::None)
+      Qualifiers.push_back(model::Qualifier::Qualifier::createConst());
+
+    if (Qualifiers.size()) {
+      model::QualifiedType Qty(ProcessedTypes[RefTy], Qualifiers);
+      TypeDef->UnderlyingType = Qty;
+
+      auto TyPath = Model->recordNewType(std::move(TypedefTy));
+      ProcessedTypes[CurrentTypeIndex] = TyPath;
+    }
+  } else {
+    revng_log(DILogger, "LF_MODIFIER: Unknown ref type " << RefTy.getIndex());
+  }
+
+  return Error::success();
+}
+
+// Parse LF_MEMBER.
+Error PDBImporterTypeVisitor::visitKnownMember(CVMemberRecord &Record,
+                                               DataMemberRecord &Member) {
+  InProgressMemberTypes[CurrentTypeIndex].push_back(Member);
```

Comment:
So you accumulate them and process them all at the end?

Reply:
Yes. It is being processed when we encounter a `struct type` represented by the `CurrentTypeIndex`.